lua/lazyvim/plugins/extras/lang/sql.lua

```diff
@@ -0,0 +1,55 @@
+local sql_ft = { "sql", "mysql", "plsql" }
+
+return {
+  {
+    "nvim-treesitter/nvim-treesitter",
+    optional = true,
+    opts = function(_, opts)
+      if type(opts.ensure_installed) == "table" then
+        vim.list_extend(opts.ensure_installed, { "sql" })
+      end
+    end,
+  },
+  {
+    "folke/edgy.nvim",
+    optional = true,
+    opts = function(_, opts)
+      table.insert(opts.right, {
+        title = "Database",
+        ft = "dbui",
+        pinned = true,
+        open = function()
+          vim.cmd("DBUI")
+        end,
+      })
+
+      table.insert(opts.bottom, {
+        title = "DB Query Result",
+        ft = "dbout",
+      })
+    end,
+  },
+  {
+    "tpope/vim-dadbod",
+    cmd = { "DBUI", "DBUIToggle", "DBUIAddConnection", "DBUIFindBuffer" },
+    dependencies = {
+      "kristijanhusak/vim-dadbod-ui",
+      { "kristijanhusak/vim-dadbod-completion", ft = sql_ft },
+      { "jsborjesson/vim-uppercase-sql", ft = sql_ft },
+    },
+    init = function()
+      vim.g.db_ui_save_location = vim.fn.stdpath("data") .. "/db_ui"
+      vim.g.db_ui_use_nerd_fonts = true
+      vim.g.db_ui_execute_on_save = false
+      vim.g.db_ui_use_nvim_notify = true
+
+      vim.api.nvim_create_autocmd("FileType", {
+        pattern = sql_ft,
+        callback = function()
+          ---@diagnostic disable-next-line: missing-fields
+          require("cmp").setup.buffer({ sources = { { name = "vim-dadbod-completion" } } })
```

Comment:
This will make other completion engines unavailable which you might want to keep, effectively breaking completion for e.g. snippets, buffer text or keywords.

See this discussion: https://github.com/kristijanhusak/vim-dadbod-completion/issues/53

It would be desirable to _extend_ the list of completion engines with `"vim-dadbod-completion"` rather than hardcode it into becoming the only one. I'm unfortunately not sure how/if it's possible... and I asked about it in the above linked discussion.

This is what my setup currently looks like: https://github.com/fredrikaverpil/dotfiles/blob/5b2a1bcb793e60a60f6aeb3f94bb3c758c9d4b02/nvim-lazyvim/lua/plugins/sql.lua